source/dataman/CLS/CLSDbUpgrade1Pt2.cpp

```diff
@@ -131,3 +180,26 @@ QString CLSDbUpgrade1Pt2::description() const
 {
 	return QString("Check the AMDBThumbnail table on the thumbnails of AMFacility table.");
 }
+
+QMap<int, QString> CLSDbUpgrade1Pt2::queryAMFacilityTableThumbnails() const
+{
+	QMap<int, QString> facilityTableThumbnails;
+
+	QSqlQuery query = databaseToUpgrade_->query();
+	query.prepare(QString("SELECT id, title FROM %1 where objectTableName='%2'").arg(thumbnailTableName_).arg(facilityTableName_));
+
+	if (query.exec()) {
+		while (query.next()) {
+			facilityTableThumbnails.insert(query.value(0).toInt(), query.value(1).toString());
+		}
+	} else {
+
+		AMErrorMon::alert(this, CLSDbUpgrade1Pt2_FAIL_TO_QUERY_FACILITY_THUMBNAIL, QString("%1").arg(query.lastQuery()));
+		AMErrorMon::alert(this, CLSDbUpgrade1Pt2_FAIL_TO_QUERY_FACILITY_THUMBNAIL, QString("Query to find objectTableName (%1) in table %2 is failed.").arg(facilityTableName_).arg(thumbnailTableName_));
+	}
+
+//	query.finish();
```

Comment:
What I meant was I think that commented line is necessary.  You should always call query.finish() if you are done with an QSqlQuery.